src/main/java/pl/wojciechkostecki/revel/service/LocalService.java

```diff
@@ -0,0 +1,45 @@
+package pl.wojciechkostecki.revel.service;
+
+import org.springframework.stereotype.Service;
+import pl.wojciechkostecki.revel.model.Local;
+import pl.wojciechkostecki.revel.repository.LocalRepository;
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+
+@Service
+public class LocalService {
+    private final LocalRepository localRepository;
+
+    public LocalService(LocalRepository localRepository) {
+        this.localRepository = localRepository;
+    }
+
+    public Local save(Local local) {
+        return localRepository.save(local);
+    }
+
+    public List<Local> getAll() {
+        return localRepository.findAll();
+    }
+
+    public Local findById(Long id) {
+        return localRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(String.format("Couldn't find a local with id: %s", id)));
+    }
+
+    public List<Local> findByName(String name) {
+        return localRepository.findByNameContainingIgnoreCase(name);
+    }
+
+    public Local updateLocal(Long id, Local local) {
+        if(id == null){
+            throw new NullPointerException(String.format("No local with the given id %s", id));
```

Comment:
That doesnt make sense. You are checking if id in path is null? You wouldn't get request sent to this endpoint if it would be null. Also exception thrown doesn't make sense.  You would get NPE thrown but it wouldn't be yours. You would get it because of String.format and trying to put in `id` which is null according to your if statement.